cmd/telemetry-agent/main.go

```diff
@@ -164,12 +164,34 @@ func processMetrics(ctx context.Context, c config.Config, platformClient *platfo
 
 		metricsLogger := l.With(zap.String("file", pillarM.Filename))
 		platformCtx := platformLogger.GetContextWithLogger(ctx, metricsLogger.Desugar())
-		// send request to Percona Platform
-		if err := platformClient.SendTelemetry(platformCtx, "", report); err != nil {
-			if !errors.Is(err, context.Canceled) {
-				metricsLogger.Errorw("error during sending telemetry, will try on next iteration", zap.Error(err))
+		err := func() error {
+			funcCtx, cancel := context.WithTimeout(platformCtx, 60*time.Second)
+			defer cancel()
+			// send request to Percona Platform
+			return platformClient.SendTelemetry(funcCtx, "", report)
+		}()
+		if err != nil {
```

Comment:
This condition can be one more `case` in switch

Reply:
em, not sure I got your idea

Reply:
Nevermind. I thought it will look better if one of the cases will be like `case err == nil: // do nothing`, but it's not.